src/rulesets/Base/mapreduce.jl

```diff
@@ -194,6 +219,7 @@ function ∇prod_dims(vald::Val{dims}, x, dy, y=prod(x; dims=dims)) where {dims}
     ∇prod_dims!(dx, vald, x, dy, y)
     return dx
 end
+∇prod_dims(::Val{dims}, x, dy::AbstractZero, y=prod(x; dims=dims)) where {dims} = dy
```

Comment:
Is this meant to be part of this PR?

Reply:
It's here because I found out it was missing when writing a test for this PR. Could be done separately I guess.